tcms/core/management/commands/set_permissions.py

```diff
@@ -0,0 +1,115 @@
+from django.contrib import auth
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group, Permission
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = 'Set permissions for Testers group and remove stale ones.'
+
+    @staticmethod
+    def get_all_permissions():
+        """
+            Returns a set of all available permissions in format
+            app_label.perm_codename
+            https://github.com/timonweb/django-debug-permissions
+        """
+        permissions = set()
+
+        # We create (but not persist) a temporary superuser and use it to game the
+        # system and pull all permissions easily.
+        tmp_superuser = get_user_model()(
+            is_active=True,
+            is_superuser=True
+        )
+
+        # We go over each AUTHENTICATION_BACKEND and try to fetch
+        # a list of permissions
+        for backend in auth.get_backends():
+            if hasattr(backend, "get_all_permissions"):
+                # pylint: disable=objects-update-used
+                permissions.update(backend.get_all_permissions(tmp_superuser))
+
+        # Make an unique list of permissions sorted by permission name.
+        return permissions
+
+    @staticmethod
+    def get_all_installed_apps():
+        """
+            Returns a set of currently installed applications as app_labels
+        """
+        applications = []
+        for app_config in apps.get_app_configs():
+            applications.append(app_config.label)
+        return applications
+
+    @staticmethod
+    def kiwi_apps():
+        """
+            Returns a list of Kiwi TCMS related applications with permissions
+            to be assigned to Tester group
+        """
+        kiwi_apps = ['bugs', 'django_comments', 'linkreference', 'management',
+                     'testcases', 'testplans', 'testruns', 'attachments']
+        return kiwi_apps
+
+    @staticmethod
+    def permissions_exclude():
+        """
+            Returns a list of permissions NOT to be assigned to Tester group
+        """
+        perm_exclude = ['attachments.delete_foreign_attachments', ]
+        return perm_exclude
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--auto', '--noinput', '--no-input', action='store_false',
+            dest='interactive',
+            help='Automatic mode. Does not require user confirmation',
+        )
+
+        parser.add_argument(
+            '--quiet', action='store_false', dest='verbose',
+            help='Suppress most of the console output',
+        )
+
+    def handle(self, *args, **kwargs):
+        all_permissions = self.get_all_permissions()
+        installed_apps = self.get_all_installed_apps()
+        kiwi_apps = self.kiwi_apps()
+        perm_exclude = self.permissions_exclude()
+
+        # Add all permissions to Testers group
+        tester = Group.objects.get(name='Tester')
```

Comment:
assign_default_group_permissions() assigns permissions if the group has no permissions at all (`if tester.permissions.count() == 0:`). I think we'd rather add just missing permissions for Kiwi TCMS applications.

Reply:
You are duplicating functionality here. Adjust the function so it can be used in two different scenarios. A default parameter whether or not to check for existing permissions will do the job.

Reply:
Should I rework my function so it could to be later included in `tcms/utils/permissions.py` replacing `assign_default_group_permissions()`?